external_resources/README.md

```diff
@@ -1,48 +1,162 @@
 # External Resource Availability Workflow
 
-This document describes the workflow for the external resources validation tasks.
+This document describes the workflow for validating external resources and integrating with the Wayback Machine.
 
 **SECTIONS**
 
 1. [Overview](#overview)
-1. [Enabling Task](#enabling-task)
+1. [Enabling Tasks](#enabling-tasks)
+1. [Wayback Machine Integration](#wayback-machine-integration-1)
+1. [Wayback Machine Removal Requests](#wayback-machine-removal-requests)
 1. [Frequency Control](#frequency-control)
 1. [Rate Limiting](#rate-limiting)
 1. [Task Priority](#task-priority)
-
+1. [Management Commands](#management-commands)
+1. [Code References](#code-references)
 
 # Overview
 
-This assumes that celery beat scheduler is installed and enabled, which is required for the task scheduling.
+This assumes that **Celery beat scheduler** is installed and enabled, which is required for the task scheduling.
+
+The frequency for **external resource checking** is set to once per week (`1/week`). All external resources, both new and existing, are validated weekly, regardless of their last status.
+
+The **Wayback Machine integration** is enabled by default. Valid external resources are submitted for archiving as part of the external resource validation task. The system monitors the submission status and avoids resubmitting resources within a specified interval. The status of Wayback Machine archiving jobs is updated every 6 hours to track the progress of pending jobs.
+
+If a resource is valid, the `check_external_resources` task triggers the `submit_url_to_wayback_task` to archive it.
+
+Here’s a high-level description of the process:
+
+### External Resource Validation:
+
+- Task is automatically added in scheduler on system start.
+- On execution, all available external resources are retrieved from DB.
+- Gathered data is divided into preconfigured batch sizes, defined by `BATCH_SIZE_EXTERNAL_RESOURCE_STATUS_CHECK` in [websites/constants.py](/websites/constants.py), which is currently set to `100`.
+- All batches are grouped into a single Celery task and executed.
+- Each batch-task iterates over batch to validate availability of each resource and its backup resource if available.
+- The status of resource is then added to DB.
+- Batch tasks have a preconfigured rate limiter and lower priority by default.
+
+### Wayback Machine Integration:
+
+- When external resource validation occurs, valid external resources are submitted to the Wayback Machine for archiving.
```

Comment:
Does this mean that we could possibly submit _all_ the external resources to the Wayback Machine _every_ week?

Reply:
We could technically submit all the external resources every week, but we are not doing this right now, as the `30` days `WAYBACK_SUBMISSION_INTERVAL_DAYS` variable prevents this.

Also, even though the link checking task is set to occur every week, its priority is lowest (`4`). And the `submit_url_to_wayback_task` function it calls, its priority is also low (`3`). So they may take much longer before the ERs actually finish execution.